pontoon/base/tests/__init__.py

```diff
@@ -40,6 +40,20 @@ def client_login(self, user=None):
 
         return user
 
+    def assert_redirect_with_error(self, url, message, after_redirect_url='http://testserver/'):
+        """
+        Some of the views are handling errors by returning redirect to the main page
+        and displaying message to the user
+        """
+        response = self.client.get(url)
```

Comment:
I think it's better to pass the response into the assertion instead of passing a URL and getting the response here, so that we can test all kinds of client requests instead of just plain GETs with no arguments.